challenges/multiview_matting/worker.py

```diff
@@ -0,0 +1,284 @@
+                   if name == obj2.asset_id][0]['metadata']["category"],
+    }
+    scene.add(obj2)
+
+  distractor_dir = np.vstack(distractor_locs)
+  distractor_dir /= np.linalg.norm(distractor_dir, axis=-1, keepdims=True)
+
+  size_multiple = 1.5
+
+material = kb.PrincipledBSDFMaterial(
+    color=kb.Color.from_hsv(rng.uniform(), 1, 1),
+    metallic=1.0, roughness=0.2, ior=2.5)
+
+table = kb.Cube(name="floor",
+                scale=(obj_size*size_multiple, obj_size*size_multiple, 0.02),
+                position=(0, 0, -0.02), material=material)
+scene += table
+
+logging.info("Loading background HDRIs from %s", FLAGS.hdri_dir)
+
+hdri_source = kb.AssetSource.from_manifest(FLAGS.hdri_assets)
+
+train_backgrounds, held_out_backgrounds = hdri_source.get_test_split(
+    fraction=0.1)
+
+if FLAGS.backgrounds_split == "train":
+  logging.info("Choosing one of the %d training backgrounds...",
+               len(train_backgrounds))
+  background_hdri = hdri_source.create(asset_id=rng.choice(train_backgrounds))
+else:
+  logging.info("Choosing one of the %d held-out backgrounds...",
+               len(held_out_backgrounds))
+  background_hdri = hdri_source.create(
+      asset_id=rng.choice(held_out_backgrounds))
+dome = add_hdri_dome(scene, background_hdri)
+
+renderer._set_ambient_light_hdri(background_hdri.filename)
+
+# --- Add Klevr-like lights to the scene
+scene += kb.assets.utils.get_clevr_lights(rng=rng)
+
+def sample_point_in_half_sphere_shell(
+    inner_radius: float,
+    outer_radius: float,
+    rng: np.random.RandomState
+    ):
+  """Uniformly sample points that are in a given distance
+     range from the origin and with z >= 0."""
+
+  while True:
+    v = rng.uniform((-outer_radius, -outer_radius, obj_height/1.2),
+                    (outer_radius, outer_radius, obj_height))
+    len_v = np.linalg.norm(v)
+    correct_angle = True
+    if add_distractors:
+      cam_dir = v[:2] / np.linalg.norm(v[:2])
+      correct_angle = np.all(np.dot(distractor_dir, cam_dir) < np.cos(np.pi / 9.))
+    if inner_radius <= len_v <= outer_radius and correct_angle:
+      return tuple(v)
+
+# --- Keyframe the camera
+scene.camera = kb.PerspectiveCamera()
+for frame in range(FLAGS.frame_start, FLAGS.frame_end + 1):
+  # scene.camera.position = (1, 1, 1)  #< frozen camera
+  scene.camera.position = sample_point_in_half_sphere_shell(
+      obj_size*1.7, obj_size*2, rng)
+  scene.camera.look_at((0, 0, obj_height/2))
+  scene.camera.keyframe_insert("position", frame)
+  scene.camera.keyframe_insert("quaternion", frame)
+
+# --- Rendering
+logging.info("Rendering the scene ...")
+renderer.save_state(job_dir / "scene.blend")
+data_stack = renderer.render()
+
+# --- Postprocessing
+kb.compute_visibility(data_stack["segmentation"], scene.assets)
+data_stack["segmentation"] = kb.adjust_segmentation_idxs(
+    data_stack["segmentation"],
+    scene.assets,
+    [obj]).astype(np.uint8)
+
+# --- Discard non-used information
```

Comment:
The API has been simplified significantly here.
Please look at `examples/shapenet.py`

Reply:
This is changed. Please check.